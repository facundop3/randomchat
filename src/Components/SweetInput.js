import React, { useState } from "react";

const SweetInput = props => {
  const {
    placeholder = "",
    handleSubmit,
    cleanAfter,
    hotAction,
    handleChange,
    hotValue,
    labelText
  } = props;

  const [inputValue, setInputValue] = useState("");
  const handleInput = ({ target: { value } }) => {
    setInputValue(value);
  };
  const handleClick = ev => ev.target.focus();
  const preventDefault = ev => {
    ev.preventDefault();
    handleSubmit && handleSubmit(inputValue);
    setInputValue(cleanAfter ? "" : inputValue);
  };
  return (
    <form
      onSubmit={preventDefault}
      style={{ position: "relative" }}
      className="col-12"
    >
      <div class="input-group">
        {hotAction ? (
          <input
            type="text"
            onChange={handleChange}
            value={hotValue}
            placeholder={placeholder}
            onClick={handleClick}
            class="form-control"
            aria-label={labelText}
          />
        ) : (
          <input
            Type="text"
            onChange={handleInput}
            value={inputValue}
            placeholder={placeholder}
            onClick={handleClick}
            class="form-control"
            aria-label={labelText}
          />
        )}
      </div>
    </form>
  );
};

export default SweetInput;
