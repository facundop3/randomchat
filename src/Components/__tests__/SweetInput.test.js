import React from 'react'
import {shallow, configure} from 'enzyme'
import SweetInput from '../SweetInput'
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});


/**
 * It takes a snapshot of the component and compares with new versions of it
 * in order to avoid unexpected UI changes
 */
describe('SweetInput', ()=>{
  it('renders correctly', ()=>{
    const wrapper = shallow(<SweetInput />)
    expect(wrapper).toMatchSnapshot()
  })
})