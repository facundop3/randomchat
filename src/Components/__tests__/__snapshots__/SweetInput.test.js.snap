// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SweetInput renders correctly 1`] = `
<styled.div>
  <form
    onSubmit={[Function]}
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <FontAwesomeIcon
      border={false}
      className=""
      color="#2ca5e0"
      fixedWidth={false}
      flip={null}
      icon="paper-plane"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      style={
        Object {
          "padding": "5.5px",
          "position": "absolute",
          "right": 0,
        }
      }
      symbol={false}
      title=""
      transform={null}
    />
    <styled.input
      onChange={[Function]}
      onClick={[Function]}
      placeholder=""
      type="text"
      value=""
    />
  </form>
</styled.div>
`;
